core/src/main/scala/org/apache/spark/deploy/history/EventLogFileWriters.scala

```diff
@@ -229,6 +232,15 @@ class SingleEventLogFileWriter(
     writeLine(eventJson, flushLogger)
   }
 
+  override def writeToXAttr(attrName: String, attrValue: String): Unit = {
+    try {
+      fileSystem.setXAttr(new Path(inProgressPath), attrName, attrValue.getBytes())
+    } catch {
+      case _: IOException =>
```

Comment:
We may also need to handle `UnsupportedOperationException` , some fs do not really implement `setXAttr` interface, such as `o.a.hadoop.fs.RawLocalFileSystem`